src/Process/Runner/MutationTestingRunner.php

```diff
@@ -103,6 +112,22 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
 
                 return false;
             })
+            ->filter(function (Mutant $mutant) {
+                $mutatorName = $mutant->getMutation()->getMutatorName();
+
+                if (!array_key_exists($mutatorName, $this->ignoreSourceCodeMutatorsMap)) {
+                    return true;
+                }
+
+
+                foreach ($this->ignoreSourceCodeMutatorsMap[$mutatorName] as $sourceCodeRegex) {
+                    if (preg_match("/-(.*?){$sourceCodeRegex}/", $mutant->getDiff()) === 1) {
```

Comment:
```suggestion
                    if (preg_match("/^-(.*?){$sourceCodeRegex}/", $mutant->getDiff()) === 1) {
```

Then it has to be like this, what do you think?

Reply:
no, because `$mutant->getDiff()` returns:

```
--- Original
+++ New
@@ @@
         $this->getString();
         return 'hello';
     }
-    public function getString()
+    protected function getString()
     {
         return 'string';
     }
 }
```

so, with `^` it won't match

Reply:
Well, I'm more concerned about it matching a dash somewhere else:

https://3v4l.org/QvZ2G

Reply:
I agree with @sanmai : the current version is prone to error when dash is elsewhere.
I have the following suggestions:

1. use `^` with `m` modifier to only and exact match the original; the theoretical false positive of `--- Original` part of the diff, even if possible, is negligible
1. use `u` modifier, since UTF8 code is allowed
1. drop `(.*?)` as is prohibits exact matches to the user
1. add `$` for the same reason, to allow exact matches
1. be permissive only to indentations, so add `\s*` at start

All in all, the final form should be:

```suggestion
                    if (preg_match("/^-\s*{$sourceCodeRegex}$/mu", $mutant->getDiff()) === 1) {
```

So that user is allowed to choose between `getString` and `.*getString.*` regexes.